tex/rules.tex

```diff
@@ -270,16 +270,33 @@ \subsection{Goalie \label{ref-008}}
 
 \subsection{Pushing \label{ref-pushing}}
 
-Within the penalty area, the goalie has priority. Attacking robots are not
-supposed to push the goalie in any way.
-
-If the attacker and the goalie touch each other and at least one of them has
-physical contact with the ball, the ball will be moved to the nearest
-unoccupied neutral spot immediately.
+Robots are not allowed to push each other at any time and at any place.
+This applies to two robots of the same team as well as two robots of the competing opponents.
+The robots may use vision, touch sensors attached to avoidance rings, or any other method
+to detect the avoidance rings of other robots, see section \ref{ref-avoidance}.
+A robot is allowed to play only if it can reliably detect touching or coming close to 
+another robot in all directions of its own movement. When the robot detects another
+robot in any particular direction, it must stop all movement in that direction immediately.
+If any robot happens to push any other robot because it has failed to detect the other robot,
+it is removed from play for 30 seconds by a referee.
 
 If a goal is scored as a result of this pushed-situation, it will not be
 granted.
 
+\subsection{Pushing with ball\label{ref-pushing-with-ball}}
+
+Robots are not allowed to push each other while both pushing the same ball in the opposite directions
+even without touching each other.
+Whenever the robot comes in contact with the ball, it is allowed to push it forward as long as 
+there is no force working in the opposite direction. If a robot tries to push a ball forward, 
+but the ball is not rolling freely, it must stop moving in the direction of the ball within 
+3 seconds for at least another 3 seconds and move in another direction for at least 5 cm. 
+The referee notices this by counting aloud: 1-2-3 ; 1-2-3.
```

Comment:
Although I understand the spirit of this rule and see how it could work in theory, I have pretty big doubts about its practical feasibility.

Even without this change there are many things the referees need to keep track of on the field. Between multiple defense, pushing, damaged and out of bounds robots, I have very hard time imagining how could the referees track robots that have stopped moving in some direction, especially if a loud count would be necessary. In my view, that blocks the referee from doing anything else for at least 6 seconds -- that seems quite a lot. I also have a hard time imagining how would a referee differentiate between the robot moving less than 5cm and more than 5cm.

Would it be possible to alter this part of the proposed rule to make it more practically feasible?